src/storage/Storage.ts

```diff
@@ -90,9 +89,8 @@ export class Storage extends EventEmitter {
                 const uuid = uuidv1()
                 const timeout = setTimeout(() => {
                     this.pendingStores.delete(uuid)
-                    this.store(streamMessage)
-                        .then(resolve)
-                        .catch(reject)
+                    // eslint-disable-next-line promise/catch-or-return
+                    this.store(streamMessage).then(resolve, reject)
```

Comment:
```
// eslint-disable-next-line promise/catch-or-return
```
bad rule?

Reply:
Yeah it's not really right, better than nothing but `p.then(success).catch(fail)`:
1. creates two promises, causing evaluation to be unnecessarily delayed by an additional microtask
2. causes the `.catch` to fire if the `success` handler throws/rejects, which can lead to very confusing behaviour as it can invoke both `success` AND `fail` handlers for a single task.

`p.then(success, fail)` should generally be used by default, especially if the intent is to return control to the caller. Only `.catch` if you really need to also catch what happened in the `.then`.